helpers/kernel_config.go

```diff
@@ -174,53 +175,69 @@ var KernelConfigKeyIDToString = map[KernelConfigOption]string{
 type KernelConfig struct {
 	configs         map[KernelConfigOption]interface{} // predominantly KernelConfigOptionValue, sometimes string
 	needed          map[KernelConfigOption]interface{}
-	KConfigFilePath string
+	kConfigFilePath string
 }
 
 // InitKernelConfig inits external KernelConfig object
 func InitKernelConfig() (*KernelConfig, error) {
 	config := KernelConfig{}
 
 	// special case: user provided kconfig file (it MUST exist)
+
 	osKConfigFilePath, err := checkEnvPath("LIBBPFGO_KCONFIG_FILE") // override /proc/config.gz or /boot/config-$(uname -r) if needed (containers)
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
 	if len(osKConfigFilePath) > 2 {
 		if _, err := os.Stat(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
-		config.KConfigFilePath = osKConfigFilePath // user might need to know used KConfig file in order to override kconfig for libbpf (example)
+		config.kConfigFilePath = osKConfigFilePath
 		if err := config.initKernelConfig(osKConfigFilePath); err != nil {
-			return nil, err
+			return &config, err
 		}
 
 		return &config, nil
 	}
 
 	// fastpath: check config.gz in procfs first
+
 	configGZ := "/proc/config.gz"
 	if _, err1 := os.Stat(configGZ); err1 == nil {
-		config.KConfigFilePath = configGZ // same thing here
+		config.kConfigFilePath = configGZ
 		if err2 := config.initKernelConfig(configGZ); err2 != nil {
-			return nil, err2
+			return &config, err2
 		}
 
 		return &config, nil
 	} // ignore if /proc/config.gz does not exist
 
 	// slowerpath: /boot/$(uname -r)
+
 	releaseVersion, err := UnameRelease()
 	if err != nil {
-		return nil, err
+		return &config, err
 	}
+
 	releaseFilePath := fmt.Sprintf("/boot/config-%s", releaseVersion)
-	config.KConfigFilePath = releaseFilePath // and here
-	if err := config.initKernelConfig(releaseFilePath); err != nil {
-		return nil, err
-	}
+	config.kConfigFilePath = releaseFilePath
+	err = config.initKernelConfig(releaseFilePath)
+
+	return &config, err
+}
+
+// GetKernelConfigFilePath gives the kconfig file chosen by InitKernelConfig during initialization
+func (k *KernelConfig) GetKernelConfigFilePath() string {
+	return k.kConfigFilePath
+}
+
+// AddCustomKernelConfigs allows user to extend list of possible existing kconfigs to be parsed from kConfigFilePath
+func (k *KernelConfig) AddCustomKernelConfigs(key KernelConfigOption, value string) error {
+	// extend initial list of kconfig options: add other possible existing ones
+	KernelConfigKeyIDToString[key] = value
+	KernelConfigKeyStringToID[value] = key
 
-	return &config, nil
+	return k.initKernelConfig(k.kConfigFilePath)
```

Comment:
This is that case of .. after reviews it got weird at the end. This function accepted a map[KernelConfigOption][string] to extend all supported kernel configs with a bunch of added other options. Now, accepting a single argument.. re-reading all the file every time there is an update is weird indeed.

Reply:
Ah ok.. for this one, for the record.. we're not saving ALL existing kconfig values into existing object's memory, only the ones we have knowledge of (this was already done). So, whenever we add new KernelConfigOption's, we need to re-read the kconfig file to set its value. 

If I keep value of all existing kconfig options in the 1st parse, then there is not even a need for AddCustomKernelConfigs() function. Since I prefer not to alter this working (and tested) logic currently, I'm adding a TODO here to improve that so we don't have to parse the file multiple times. It no big deal as the file will be cached (or in procfs), but, still.. not nice.

Reply:
addressed at: https://github.com/aquasecurity/libbpfgo/pull/73 (not ideal, but better than reloading each time a new custom option is added).